scene/2d/navigation_region_2d.cpp

```diff
@@ -605,8 +610,13 @@ NavigationRegion2D::NavigationRegion2D() {
 	region = NavigationServer2D::get_singleton()->region_create();
 	NavigationServer2D::get_singleton()->region_set_enter_cost(region, get_enter_cost());
 	NavigationServer2D::get_singleton()->region_set_travel_cost(region, get_travel_cost());
+
+	NavigationServer3D::get_singleton_mut()->connect("map_changed", callable_mp(this, &NavigationRegion2D::_map_changed));
+	NavigationServer3D::get_singleton_mut()->connect("navigation_debug_changed", callable_mp(this, &NavigationRegion2D::_map_changed));
 }
```

Comment:
Again, if these are purely for debug, they can be enclosed in `#ifdef DEBUG_ENABLED` maybe?